src/Module/Manifest.php

```diff
@@ -31,22 +31,33 @@ public static function rawContent(array $parameters = [])
 	{
 		$config = DI::config();
 
-		$tpl = Renderer::getMarkupTemplate('manifest.tpl');
-
-		header('Content-type: application/manifest+json');
-
-		$touch_icon = $config->get('system', 'touch_icon', 'images/friendica-128.png');
-		if ($touch_icon == '') {
-			$touch_icon = 'images/friendica-128.png';
+		$touch_icon = $config->get('system', 'touch_icon') ?: 'images/friendica-128.png';
+
+		$theme = DI::config()->get('system', 'theme');
+
+		$manifest = [
+			'name'        => $config->get('config', 'sitename', 'Friendica'),
+			'start_url'   => DI::baseUrl()->get(),
+			'display'     => 'standalone',
+			'description' => $config->get('config', 'info', DI::l10n()->t('A Decentralized Social Network')),
+			'short_name'  => 'Friendica',
+			'icons'       => [
+				[
+					'src'   => DI::baseUrl()->get() . '/' . $touch_icon,
+					'sizes' => '128x128',
```

Comment:
Am I a nitpicker when I just point out that custom icons don't necessarily are in "128x128"? :grin:

Reply:
Not necessarily, but I don't think the image size deliberation should happen when we call the manifest, rather when we save the touch icon setting.